lib/trip_dispatcher.rb

```diff
@@ -27,14 +36,50 @@ def inspect
               #{passengers.count} passengers>"
     end
 
+    def request_trip(passenger_id)
+      available_drivers = @drivers.select {|driver|driver.status == :AVAILABLE}
+    
+      driver = nil
+      oldest_endtime = Time.now
+
+      available_drivers.each do |indiv_driver|
```

Comment:
Good work implementing this logic. You may consider encapsulating this logic in a helper method.